composeApp/src/commonMain/kotlin/com/github/familyvault/ui/screens/main/MainScreen.kt

```diff
@@ -112,43 +105,112 @@ class MainScreen : Screen {
+                        startImagePicker = false
+                    }
+                }
+            }
+            1 -> {
+                FloatingActionButton(onClick = {
+                    startDocumentPicker = true
+                }) {
+                    Icon(
+                        Icons.Filled.UploadFile, 
+                        stringResource(Res.string.file_cabinet_upload),
+                    )
                 }
 
-                startPicker = false
+                if (startDocumentPicker) {
+                    LaunchedEffect(Unit) {
+                        try {
+                            val documents = documentPicker.pickDocumentsAndReturnByteArrays()
+
+                            if (documents.isNotEmpty()) {
+                                isUploading = true
+                                
+                                withContext(Dispatchers.IO) {
+                                    val documentUrls = documentPicker.getSelectedDocumentUrls()
+
+                                    documentUrls.forEachIndexed { index, uri ->
+                                        val documentName = documentPicker.getDocumentNameFromUri(uri) ?: "document_$index.pdf"
+                                        val documentMimeType = documentPicker.getDocumentMimeTypeFromUri(uri) ?: "application/pdf"
+
+                                        documents.getOrNull(index)?.let { docBytes ->
+                                            fileCabinetService.sendDocumentToFamilyGroupStore(
+                                                docBytes,
+                                                documentName,
+                                                documentMimeType
+                                            )
+                                        }
+                                    }
+                                }
+                                
+                                isUploading = false
+                            } else {
+                            }
+                        } catch (e: Exception) {
+                            errorMessage = "Error: ${e.message}"
+                        } finally {
+                            startDocumentPicker = false
+                        }
+                    }
+                }
             }
         }
 
         if (isUploading) {
             CircularProgressIndicatorDialog(stringResource(Res.string.file_cabinet_sending_files))
         }
+        
+        errorMessage?.let { error ->
+            AlertDialog(
+                onDismissRequest = { errorMessage = null },
+                title = { Text("Error") },
+                text = { Text(error) },
+                confirmButton = {
+                    TextButton(
+                        onClick = { errorMessage = null }
+                    ) {
+                        Text("OK")
+                    }
+                }
+            )
+        }
     }
```

Comment:
Wyrzućmy ten kod do jakiegoś innego komponentu, bo zaczyna się tutaj robić potwór. 

I jak wyrzucisz to do innego komponentu, to też podziel na Images i Documents. Będzie prościej tym zarządzać.